### ISSUE TITLE
Group Level Injection RESV Control Produces Confusing Results

### ISSUE THREAD

**Issue by bska** [OPM/opm-simulators#3347]
I came across some unexpected behaviour whilst analyzing an unrelated problem.  It **appears** that the simulator's implementation of `GCONINJE`'s `RESV` control applies unit conversion (or formation volume factor reduction) one too many times, but I've not been able to pinpoint the exact source of the issue.

I changed [SPE1CASE2](https://github.com/OPM/opm-tests/blob/d7a6d647214afe5b52dd3ecd254b17c9c9001edc/spe1/SPE1CASE2.DATA) as follows
```diff
RPTSCHED
@@ -411,8 +429,14 @@ WCONPROD
 
 WCONINJE
 -- Item #:1     2       3       4      5      6  7
-       'INJ'   'GAS'   'OPEN'  'RATE'  100000 1* 9014 /
+--     'INJ'   'GAS'   'OPEN'  'RATE'  100000 1* 9014 /
+  'INJ'  'GAS'  'OPEN'  'GRUP'  1* 1* 9014 /
+/
+
+GCONINJE
+  'G1' 'GAS' 'RATE' 100.0E3 60.0E3 /
 /
```
(in addition to outputting a few more group-level summary vectors).  The `60.0E3` RB/day limit (item 5; roughly 0.1104 m3/s) is a sort of average value of the well-level reservoir volume injection rate in the standard run.  I get the reservoir volume injection rate results in the picture below.  The blue curve (left-hand axis) is from a run **without** the RESV limit (i.e., with only the RATE limit).  The red curve (right-hand axis) is from a run that does include the RESV limit (60e3 RB/day).  While we should not expect the two curves to be equal since the controls differ, they should have comparable orders of magnitude.  Here, however, the run **without** RESV produces rates on the order of 50e3 RB/day to 85e3 RB/day while the run **with** the RESV constraint produces rates on the order of 200 RB/day to 620 RB/day.  As another experiment, I disabled the group constraints and placed the RESV constraint of 60e3 RB/day at the well level instead.  Doing so produces well (and group) level reservoir volume injection rates in the order of 60e3 RB/day for all times.  That suggests that the RESV problem is limited to the GCONINJE context rather than being a general problem with Flow's overall constraint implementation.

Like I said, I've tried to trace the logic here but now I'm at a point where I don't know enough about the system to know where to look anymore.  Any suggestions would be most welcome.

---

![gconinje_resv_spe1](https://user-images.githubusercontent.com/1871438/120644549-c0d0e500-c477-11eb-9a92-41663ef601aa.png)

**Comment by bska** [comment]
As a quick follow-up, I plotted the group-level surface gas volume injection rate and got a constant, numerical value of `336.8750` which is also unreasonably low for a gas rate in this context.  As an experiment I then multiplied this by `178.1076` (conversion factor for the gas formation volume factor Bg in FIELD units) and recovered the constant value `60e3` as shown in the picture below.

I don't know how to interpret that except that this points to some kind of data misinterpretation somewhere.

---

![gconinje_ggir_resv_spe1](https://user-images.githubusercontent.com/1871438/120648891-6ede8e00-c47c-11eb-848f-28604d0f747a.png)

**Comment by totto82** [comment]
The rates for GCONINJE is not scaled for RESV. See https://github.com/OPM/opm-simulators/blob/42d1b4c4a80740089439f2b4972ba3ee02c90aee/opm/simulators/wells/TargetCalculator.hpp#L87
I can make a fix tomorrow.

**Comment by bska** [comment]
> The rates for GCONINJE is not scaled for RESV

I see.  Thanks for the quick response.

> I can make a fix tomorrow.

Very good.  I'll defer this task to you as I don't know how to make the appropriate change myself.

**Comment by bska** [comment]
Fixed in #3363.
